services/supervisor/src/api.rs

```diff
@@ -12,6 +12,578 @@
+//!
+//! Proposes the new configuration for the Exonum blockchain.
+//!
+//! Configuration changes may include the following items:
+//!
+//! - Request to change the consensus configuration.
+//! - Request to start a new service instance.
+//! - Request to stop an existing service instance.
+//! - Request to resume a previously stopped service.
+//! - Request to change the configuration of an existing service instance.
+//!
+//! Configuration proposal does not cause the configuration change itself, instead it
+//! initializes a voting process: if node administrators of the network agree on the
+//! suggested proposal, the configuration is applies. Otherwise, no changes in the
+//! network configuration are performed.
+//!
+//! Voting for a configuration is performed via [`confirm-config`](#vote-for-configuration-proposal)
+//! endpoint.
+//!
+//! For more details on configuration proposals, see [crate documentation](../index.html).
+//!
+//! **Warning:** `ConfigPropose` structure should be serialized using corresponding protobuf message,
+//! and represented as a hexadecimal string.
+//!
+//! [`ConfigPropose`]: ../struct.ConfigPropose.html
+//! [`Hash`]: https://docs.rs/exonum-crypto/latest/exonum_crypto/struct.Hash.html
+//!
+//! ```
+//! use exonum::crypto::Hash;
+//! # use exonum::helpers::Height;
+//! use exonum_rust_runtime::ServiceFactory;
+//! use exonum_supervisor::{ConfigPropose, Supervisor};
+//! use exonum_testkit::{ApiKind, TestKitBuilder};
+//!
+//! # fn main() -> Result<(), failure::Error> {
+//! let mut testkit = TestKitBuilder::validator()
+//!     .with_rust_service(Supervisor)
+//!     .with_artifact(Supervisor.artifact_id())
+//!     .with_instance(Supervisor::simple())
+//!     .create();
+//!
+//! let proposal: ConfigPropose = // Proposal creation skipped...
+//! # ConfigPropose::new(0, Height(0));
+//!
+//! // In this example, query is serialized to hexadecimal string automatically.
+//! let tx_hash: Hash = testkit
+//!     .api()
+//!     .private(ApiKind::Service("supervisor"))
+//!     .query(&proposal)
+//!     .post("propose-config")?;
+//!
+//! // Create a block, so the proposal transaction will appear in the blockchain.
+//! let block = testkit.create_block();
+//!
+//! // Verify that transaction was executed successfully.
+//! assert!(block[tx_hash].status().is_ok());
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! ## Vote for Configuration Proposal
+//!
+//! | Property    | Value |
+//! |-------------|-------|
+//! | Path        | `/api/supervisor/v1/confirm-config` |
+//! | Method      | POST   |
+//! | Query type  | [`DeployRequest`] |
+//! | Return type | [`Hash`] |
+//!
+//! Votes for a pending configuration with a certain hash.
+//!
+//! Depending on the supervisor operating mode, it may be required to vote by majority of
+//! nodes (in "decentralized" mode), or one vote will be enough (in "simple" mode).
+//!
+//! The first vote is sent automatically by the node which broadcast the proposal, there is
```

Comment:
Nit: I'd rephrase "The node that broadcast the proposal is considered to have voted for it, there is no need..." because the node does not really *send* the vote, does it?